core/templates/pages/exploration-player-page/services/player-position.service.ts

```diff
@@ -40,6 +40,12 @@ export class PlayerPositionService {
     this.onChangeCallback = callback;
   }
 
+  /**
+   * Get the name of the current state.
+   * @return {object} a string that shows the name of the current state.
+   * @throws Will throw error if the index of card
+   *  is out of the range of the transcript.
```

Comment:
Hm, not sure if this is the error that will be thrown -- we don't know what error the catch block is expected to catch. Maybe we can remove this and the try-catch if it is no longer required.

@bansalnitish (author) do we still need this try-catch?

Reply:
How about keeping the try-catch just for now? If it throws an error it might be a headache to find out.

Reply:
I think it was added to debug some error in the past. If the cause of the error has been identified and fixed, we no longer require additional logging. If that is the case, we don't need the try-catch, right?

Reply:
Alright, that makes perfect sense.

Reply:
Hi @tianqi-wu, sorry if I wasn't clear earlier, but we need to first confirm if the try-catch is required or not before removing it. @bansalnitish, who is the author of the commit that introduced the try-catch, can confirm this.

Also, just to check, in your last commit 34948e7c65687a65f178d059c114eacda5823a18, it looks like the frontend tests have failed because of low coverage. The frontend tests are expected to execute automatically on your local machine when pushing to your branch -- it will prevent pushing if the test fails. So can you please confirm if this test ran on your local machine or not? Thanks!

Reply:
I agree with @kevintab95 -- I don't think we need this try-catch now!

Reply:
> Hi @tianqi-wu, sorry if I wasn't clear earlier, but we need to first confirm if the try-catch is required or not before removing it. @bansalnitish, who is the author of the commit that introduced the try-catch, can confirm this.
> 
> Also, just to check, in your last commit [34948e7](https://github.com/oppia/oppia/commit/34948e7c65687a65f178d059c114eacda5823a18), it looks like the frontend tests have failed because of low coverage. The frontend tests are expected to execute automatically on your local machine when pushing to your branch -- it will prevent pushing if the test fails. So can you please confirm if this test ran on your local machine or not? Thanks!

There wasn't a prevent pushing thing; I run tests locally on my dev server.

Reply:
Also, it's weird that once I added the try-catch statement, it can pass the frontend tests correctly, yet it doesn't when I removed it.

Reply:
It seems to fail due to a coverage issue. You can try running this command to debug what the issues is:
`python -m scripts.run_frontend_tests --check_coverage`

Reply:
> There wasn't a prevent pushing thing; I run tests locally on my dev server.

This is weird. The lint tests and the frontend tests are supposed to run when you push your changes. What command do you use to push?

Reply:
> It seems to fail due to a coverage issue. You can try running this command to debug what the issues is:
> `python -m scripts.run_frontend_tests --check_coverage`

Strangely, the modified version without try-catch statement works perfectly on my laptop. Is it possible that the tests are flaky for other reasons?

Reply:
Hi @tianqi-wu, can you confirm the command you used to push your code. For the coverage failure, you may have to write a unit test to cover the code you had modified -- you can find some useful info on writing a unit test [here](https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide). Do let me know if you need any help with that. Thanks!

Reply:
I think I just used git push for modifications; is that OK?

Reply:
Also, no matter how many tests I run locally, the coverage for this specific file seems perfectly good.

Reply:
> Hi @tianqi-wu, can you confirm the command you used to push your code. For the coverage failure, you may have to write a unit test to cover the code you had modified -- you can find some useful info on writing a unit test [here](https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide). Do let me know if you need any help with that. Thanks!

I think I find the issue.

player-position.service.ts seems to be fully covered! Before removing it manually from the blacklist in the file scripts/check_frontend_test_coverage.py, please make sure you've followed the unit tests rules correctly on: https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide#rules

This is from the frontend test; it seems this file is fully covered, but was put on a blacklist for some reasons.

Reply:
> I think I just used git push for modifications; is that OK?

As long as you do not use --no-verify while pushing, it is fine.

> player-position.service.ts seems to be fully covered! Before removing it manually from the blacklist in the file scripts/check_frontend_test_coverage.py, please make sure you've followed the unit tests rules correctly on: https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide#rules

Oh great! In that case, you can go ahead and remove it from the blacklist in `scripts/check_frontend_coverage.py`. Maybe the lines that were not covered earlier would have been the "catch" portion of the "try-catch" block -- which you have removed now.